dashboard/app/controllers/lti/v1/account_linking_controller.rb

```diff
@@ -7,6 +7,14 @@ class AccountLinkingController < ApplicationController
 
       # GET /lti/v1/account_linking/landing
       def landing
+        lti_provider = session.dig(:lms_landing, :lti_provider_name) || params[:lti_provider]
+        new_cta_type = session.dig(:lms_landing, :new_cta_type) || params[:new_cta_type]
+        user_type = session.dig(:lms_landing, :user_type) || current_user&.user_type
+
+        if lti_provider.nil? || new_cta_type.nil? || user_type.nil?
```

Comment:
Just to double-check: Is it acceptable for `lti_provider`, `new_cta_type`, or `user_type` to be an empty string?
If not, it would be better to check for `#blank?`:
```suggestion
        if lti_provider.blank? || new_cta_type.blank? || user_type.blank?
```

Reply:
Good catch, I've updated this to use `.blank?`